cmd/preflight/cmd/check_container.go

```diff
@@ -74,23 +77,41 @@ func checkContainerRunE(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("invalid configuration: %w", err)
 	}
 
-	cfg.Image = containerImage
-	cfg.ResponseFormat = formatters.DefaultFormat
+	artifactsWriter, err := artifacts.NewFilesystemWriter(artifacts.WithDirectory(cfg.Artifacts))
+	if err != nil {
+		return err
+	}
+
+	// Add the artifact writer to the context for use by checks.
+	ctx = artifacts.ContextWithWriter(ctx, artifactsWriter)
 
-	checkContainer, err := lib.NewCheckContainerRunner(ctx, cfg, submit)
+	formatter, err := formatters.NewByName(formatters.DefaultFormat)
 	if err != nil {
 		return err
 	}
 
+	opts := generateContainerCheckOptions(cfg)
+
+	checkcontainer := container.NewCheck(
+		containerImage,
+		opts...,
+	)
+
+	pc := lib.NewPyxisClient(ctx, cfg.CertificationProjectID, cfg.PyxisAPIToken, cfg.PyxisHost)
```

Comment:
The order of this creates a bug where we can no longer test in lower envs. NewCheck can either take `pc` or at the very least `cfg.PyxisHost` and have NewCheck no longer default to the default pyxis host

Reply:
Oh! I think the problem here is that I never actually pass the values to the containerCheck. I believe I already added the options, I just didn't call them. I'll double check. Good catch

Reply:
Should be resolved. I'll let you resolve this conversation when confirmed.

Reply:
I retested this on Friday and it is fixed. Thanks!